learning/prompts/documentation-lookup/45_spark_runner.md

```diff
@@ -0,0 +1,188 @@
+* Built-in metrics reporting using Apache Spark’s [metrics system](https://spark.apache.org/docs/latest/monitoring.html#metrics), which also reports Beam Aggregators.
+* Native support for Apache Beam side inputs via Apache Spark’s [broadcast variables](https://spark.apache.org/docs/latest/rdd-programming-guide.html#broadcast-variables).
+
+There are three types of Spark Runners available:
+1. A legacy Spark Runner that exclusively supports Java (and other JVM-based languages), based on Apache Spark’s [RDD and DStream](https://spark.apache.org/docs/latest/api/scala/org/apache/spark/rdd/RDD.html).
+2. A Structured Streaming Spark Runner that exclusively supports Java (and other JVM-based languages), based on the Apache Spark [Datasets](https://spark.apache.org/docs/latest/sql-programming-guide.html#datasets-and-dataframes) and [Structured Streaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html) framework. Currently, it only supports batch mode, and its coverage of the Apache Beam model is partial.
+3. A portable Spark Runner that supports Java, Python, and Go.
+
+For applications solely using Java, one of the Java-based runners is recommended. For Python or Go pipelines with Beam on Spark, opt for the portable Runner.
+
+The Spark Runner can execute Spark pipelines similar to a native Spark application, allowing deployment as a self-contained application for local mode, running on [Spark Standalone Resource Manager (RM)](https://spark.apache.org/docs/latest/spark-standalone.html), or using [YARN](https://spark.apache.org/docs/latest/running-on-yarn.html) or [Mesos](https://spark.apache.org/docs/latest/running-on-mesos.html).
+
+To execute your Apache Beam pipeline on a Spark Standalone RM, follow these steps:
+
+***Java-based Non-portable Spark Runners (Java Only)***
+
+***1. Specify Dependencies:***
+
+In the `pom.xml` file of your Java project directory, specify your dependency on the latest version of the Spark Runner:
+
+```java
+<dependency>
+  <groupId>org.apache.beam</groupId>
+  <artifactId>beam-runners-spark-3</artifactId>
+  <version>2.54.0</version>
+</dependency>
+```
+
+***2. Deploy Spark with Your Application:***
+
+When running pipelines in a Spark Standalone mode, ensure that your self-contained application includes Spark dependencies explicitly in your `pom.xml` file:
+
+```java
+<dependency>
+  <groupId>org.apache.spark</groupId>
+  <artifactId>spark-core_2.12</artifactId>
+  <version>${spark.version}</version>
+</dependency>
+
+<dependency>
+  <groupId>org.apache.spark</groupId>
+  <artifactId>spark-streaming_2.12</artifactId>
+  <version>${spark.version}</version>
+</dependency>
+```
+
+Additionally, shade the application JAR using the Maven shade plugin by executing the following code:
+
+```java
```

Comment:
@dariabezkorovaina I would just provide a list of required steps and point to documentation. This looks like too m,uch information to grasp in a single response.